docs/src/rest-api.md

```diff
@@ -0,0 +1,96 @@
+# REST API Reference
+
+REST API for programmatic access to the faucet service.
+
+## Authentication
+
+### API Keys
```

Comment:
I would move the "Authentication" section to the end. Also, it feels like the section is really not so much about authentication but about rate limiting - so, maybe we should make this the title? And then "API keys" becomes a sub-section in the "Rate limiting" section.